internal/rpcserver/rpcserver.go

```diff
@@ -6358,11 +6358,14 @@ func (s *Server) route(ctx context.Context) *http.Server {
 			return
 		}
 		ws.SetPingHandler(func(payload string) error {
-			err := ws.WriteControl(websocket.PongMessage, []byte(payload),
-				time.Now().Add(time.Second))
 			log.Debugf("ping received: len %d", len(payload))
 			log.Tracef("ping payload: %s", payload)
-			if err != nil {
+			var netErr net.Error
+			err := ws.WriteControl(websocket.PongMessage, []byte(payload),
+				time.Now().Add(websocketPongTimeout))
+			if err != nil && !errors.Is(err, websocket.ErrCloseSent) &&
+				!(errors.As(err, &netErr) && netErr.Temporary()) {
```

Comment:
Temporary is being deprecated.  Advice is to use Timeout I believe
https://pkg.go.dev/net@master#Error

Reply:
Good point.  It looks like the `errWriteTimeout` implementation in gorilla does set the timeout flag and implement it properly, so I don't think using that instead will be an issue.  I'm going to test it out with the change though before updating this since I had already tested with the existing code.

Reply:
Tested and updated.